projects/core/src/event/event.service.ts

```diff
@@ -47,29 +48,42 @@ export class EventService {
    *
    * @returns a teardown function which unregisters the given event source
    */
-  register<T>(eventType: Type<T>, source$: Observable<T>): () => void {
-    const eventMeta = this.getEventMeta(eventType);
-    if (eventMeta.mergingSubject.has(source$)) {
-      if (isDevMode()) {
-        console.warn(
-          `EventService: the event source`,
-          source$,
-          `has been already registered for the type`,
-          eventType
-        );
+  register<T>(eventType: AbstractType<T>, source$: Observable<T>): () => void {
+    const mergingSubjects: MergingSubject<T>[] = [];
+    let parentType = eventType;
+    while (
```

Comment:
I would expect the register method will be free from iterating through the prototype chain thing.